### ISSUE TITLE
Connect Beats to Elastic Cloud

### ISSUE THREAD

**Issue by jcamu** [elastic/cloud-on-k8s#6481]
Hello.
How to use this chart to target Elastic Cloud instance and not Reference to ECK-managed Elasticsearch instance ?
Is it expected ? Schedule ?
If not, what is the recommended solution for elastic cloud ?

**Comment by Evanjt1** [comment]
@jcamu

See https://github.com/elastic/cloud-on-k8s/blob/62f2e278e4184d1559510776dae108fe5248f766/config/crds/v1/all-crds.yaml#L2336-L2346

You can target a beat with an elasticsearch cluster not managed by ECK by creating a secret with connection details and specifying said secret in `elasticsearchRef.secretName`. Same for associating with kibana not managed by ECK.

I haven't tested with `cloud.id` or `cloud.auth` as seen [here](https://www.elastic.co/guide/en/beats/filebeat/current/configure-cloud-id.html), but those are just niceties for username, password, etc. anyway.

Caveat, I'm not an elastic employee, simply a user in a similar situation.

**Comment by thbkrkr** [comment]
Hello @jcamu,

@Evanjt1 is right and you can leverage the `elasticsearchRef.secretName` field to associate a Beat to an external Elasticsearch not managed by ECK.

`secretName` in `ref` field was implemented for the purpose of Stack Monitoring (#5240) but we designed it to be usable for all associations. We will make it generally usable for all associations (#6449).


Here is a usage example for Filebeat:

```yaml
apiVersion: v1
kind: Secret
metadata:
  name: cloud-es-ref
stringData:
  # explicit port otherwise beat uses 9200
  url: https://mon.es.europe-west2.gcp.elastic-cloud.com:443
  username: elastic
  password: REDACTED
---
apiVersion: v1
kind: Secret
metadata:
  name: cloud-kb-ref
stringData:
  url: https://mon.kb.europe-west2.gcp.elastic-cloud.com:9243
  username: elastic
  password: REDACTED
---
apiVersion: beat.k8s.elastic.co/v1beta1
kind: Beat
metadata:
  name: filebeat
spec:
  type: filebeat
  version: 8.6.1
  elasticsearchRef:
    secretName: cloud-es-ref
  kibanaRef:
    secretName: cloud-kb-ref
  config:
    ...
  daemonSet:
    ...
```

**Comment by thbkrkr** [comment]
I'm closing as we have https://github.com/elastic/cloud-on-k8s/issues/6449 to document this.
